infrastructure/network/addressmanager/addressmanager.go

```diff
@@ -5,1585 +5,225 @@
-	}
-
-	am.localAddressesLock.Lock()
-	defer am.localAddressesLock.Unlock()
-
-	addressKey := NetAddressKey(netAddress)
-	address, ok := am.localAddresses[addressKey]
-	if !ok || address.score < priority {
-		if ok {
-			address.score = priority + 1
-		} else {
-			am.localAddresses[addressKey] = &localAddress{
-				netAddress: netAddress,
-				score:      priority,
-			}
-		}
-	}
-	return nil
-}
-
-// getReachabilityFrom returns the relative reachability of the provided local
-// address to the provided remote address.
-func (am *AddressManager) getReachabilityFrom(localAddress, remoteAddress *appmessage.NetAddress) int {
-	const (
-		Unreachable = 0
-		Default     = iota
-		Teredo
-		Ipv6Weak
-		Ipv4
-		Ipv6Strong
-		Private
-	)
-
-	if !am.IsRoutable(remoteAddress) {
-		return Unreachable
-	}
-
-	if IsRFC4380(remoteAddress) {
-		if !am.IsRoutable(localAddress) {
-			return Default
-		}
-
-		if IsRFC4380(localAddress) {
-			return Teredo
-		}
-
-		if IsIPv4(localAddress) {
-			return Ipv4
-		}
-
-		return Ipv6Weak
-	}
-
-	if IsIPv4(remoteAddress) {
-		if am.IsRoutable(localAddress) && IsIPv4(localAddress) {
-			return Ipv4
-		}
-		return Unreachable
-	}
-
-	/* ipv6 */
-	var tunnelled bool
-	// Is our v6 is tunnelled?
-	if IsRFC3964(localAddress) || IsRFC6052(localAddress) || IsRFC6145(localAddress) {
-		tunnelled = true
-	}
-
-	if !am.IsRoutable(localAddress) {
-		return Default
-	}
-
-	if IsRFC4380(localAddress) {
-		return Teredo
-	}
+	key := netAddressKey(address)
+	addressEntry, ok := am.bannedAddresses[key]
 
-	if IsIPv4(localAddress) {
-		return Ipv4
-	}
-
-	if tunnelled {
-		// only prioritise ipv6 if we aren't tunnelling it.
-		return Ipv6Weak
-	}
-
-	return Ipv6Strong
-}
-
-// GetBestLocalAddress returns the most appropriate local address to use
-// for the given remote address.
-func (am *AddressManager) GetBestLocalAddress(remoteAddress *appmessage.NetAddress) *appmessage.NetAddress {
-	am.localAddressesLock.Lock()
-	defer am.localAddressesLock.Unlock()
-
-	bestReach := 0
-	var bestScore AddressPriority
-	var bestAddress *appmessage.NetAddress
-	for _, localAddress := range am.localAddresses {
-		reach := am.getReachabilityFrom(localAddress.netAddress, remoteAddress)
-		if reach > bestReach ||
-			(reach == bestReach && localAddress.score > bestScore) {
-			bestReach = reach
-			bestScore = localAddress.score
-			bestAddress = localAddress.netAddress
-		}
-	}
-	if bestAddress != nil {
-		log.Debugf("Suggesting address %s:%d for %s:%d", bestAddress.IP,
-			bestAddress.Port, remoteAddress.IP, remoteAddress.Port)
-	} else {
-		log.Debugf("No worthy address for %s:%d", remoteAddress.IP,
-			remoteAddress.Port)
-
-		// Send something unroutable if nothing suitable.
-		var ip net.IP
-		if !IsIPv4(remoteAddress) {
-			ip = net.IPv6zero
-		} else {
-			ip = net.IPv4zero
-		}
-		services := appmessage.SFNodeNetwork | appmessage.SFNodeBloom
-		bestAddress = appmessage.NewNetAddressIPPort(ip, 0, services)
+	if ok {
```

Comment:
As previously mentioned - it's idiomatic in go to have the error-case in the if.
So please, invert the condition here.